comp/trace/agent/impl/agent.go

```diff
@@ -77,7 +77,7 @@ func (c component) SetOTelAttributeTranslator(attrstrans *attributes.Translator)
 }
 
 func (c component) ReceiveOTLPSpans(ctx context.Context, rspans ptrace.ResourceSpans, httpHeader http.Header) source.Source {
```

Comment:
nit: you can add `hostFromAttributesHandler` to this func as well (ignore if you plan to do it in a follow-up PR)

Reply:
I tried to minimise the changes as I had many issues with repository cross dependencies. I'll do it in a follow-up PR.